frontend/libs/studio-components/src/components/StudioTableLocalPagination/StudioTableLocalPagination.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { forwardRef, useState } from 'react';
+import { StudioTableRemotePagination } from '../StudioTableRemotePagination';
+import type { Rows } from '../StudioTableRemotePagination';
+import { useTableSorting } from '../../hooks/useTableSorting';
+import { getRowsToRender } from '../StudioTableRemotePagination/utils';
+
+type StudioTableLocalPaginationProps = {
+  columns: Record<'accessor' | 'value', string>[];
+  rows: Rows;
+  size?: 'small' | 'medium' | 'large';
+  isSortable?: boolean;
+  pagination?: {
+    pageSizeOptions: number[];
+    pageSizeLabel: string;
+    nextButtonText: string;
+    previousButtonText: string;
+    itemLabel: (num: number) => string;
+  };
+};
+
+export const StudioTableLocalPagination = forwardRef<
+  HTMLTableElement,
+  StudioTableLocalPaginationProps
+>(({ columns, rows, isSortable = true, size = 'medium', pagination }, ref): React.ReactElement => {
+  const { pageSizeOptions, pageSizeLabel, itemLabel, nextButtonText, previousButtonText } =
+    pagination || {};
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [pageSize, setPageSize] = useState<number>(pagination ? pageSizeOptions[0] : undefined);
+
+  const { handleSorting, sortedRows } = useTableSorting(rows);
+  let rowsToRender: Rows;
+
+  if (isSortable) {
+    rowsToRender = getRowsToRender(currentPage, pageSize, sortedRows);
+  } else {
+    rowsToRender = getRowsToRender(currentPage, pageSize, rows);
+  }
+
+  if (!rowsToRender.length && (sortedRows.length || rows.length)) {
```

Comment:
Based on the two previous comments, I think we only need to check for `rowsToRender`?
```suggestion
  if (!rowsToRender) {
```

Reply:
If we do add a new prop for empty table text, then we can return the text content here instead of setting page 1.